services-camel/src/test/java/edu/unc/lib/dl/services/camel/destroyDerivatives/DestroyDerivativesRouterIT.java

```diff
@@ -0,0 +1,345 @@
+
+    private final static String LOC1_ID = "loc1";
+
+    @Before
+    public void init() {
+        initMocks(this);
+
+        TestHelper.setContentBase(baseAddress);
+
+        repositoryInitializer.initializeRepository();
+        PID contentRootPid = getContentRootPid();
+
+        AccessGroupSet testPrincipals = new AccessGroupSet("edu:unc:lib:cdr:admin");
+        agent = new AgentPrincipals("testUser", testPrincipals);
+
+        ContentRootObject contentRoot = repoObjLoader.getContentRootObject(contentRootPid);
+        adminUnit = repoObjectFactory.createAdminUnit(new AclModelBuilder("Unit")
+                .addUnitOwner(agent.getUsernameUri())
+                .model);
+
+        collection = repoObjectFactory.createCollectionObject(null);
+
+        contentRoot.addMember(adminUnit);
+        adminUnit.addMember(collection);
+
+        reset(destroyCollectionSrcImgProcessor);
+        reset(destroySmallThumbnailProcessor);
+        reset(destroyLargeThumbnailProcessor);
+        reset(destroyAccessCopyProcessor);
+        reset(destroyFulltextProcessor);
+
+        treeIndexer = new RepositoryObjectTreeIndexer(queryModel, fcrepoClient);
+        premisLoggerFactory.setBinaryTransferService(transferService);
+
+        when(pathFactory.getPath(any(PID.class))).thenReturn(path);
+        when(path.toNamePath()).thenReturn("path/to/object");
+        when(path.toIdPath()).thenReturn("pid0/pid1/pid2/pid3");
+    }
+
+    @Test
+    public void destroyImageTest() throws Exception {
+        WorkObject work = repoObjectFactory.createWorkObject(null);
+        FileObject fileObj = addFileToWork(work, "image/png");
+        work.addMember(fileObj);
+
+        treeIndexer.indexAll(baseAddress);
+
+        initMarkForDeletionJob(fileObj.getPid());
+        deletionJob.run();
+        initializeDestroyJob(Collections.singletonList(fileObj.getPid()));
+        destroyJob.run();
+
+        verify(destroySmallThumbnailProcessor).process(any(Exchange.class));
+        verify(destroyLargeThumbnailProcessor).process(any(Exchange.class));
+        verify(destroyCollectionSrcImgProcessor, never()).process(any(Exchange.class));
+        verify(destroyAccessCopyProcessor).process(any(Exchange.class));
+        verify(destroyCollectionSrcImgProcessor, never()).process(any(Exchange.class));
+        verify(destroyFulltextProcessor, never()).process(any(Exchange.class));
+    }
+
+    @Test
+    public void destroyCollectionImageTest() throws Exception {
+        CollectionObject collectionWithImg = repoObjectFactory.createCollectionObject(null);
+        adminUnit.addMember(collectionWithImg);
+
+        treeIndexer.indexAll(baseAddress);
+
+        // Create collection thumbnail
+        PID collPid = collectionWithImg.getPid();
+        String uuid = collPid.getUUID();
+        String binarySubPath = idToPath(uuid, HASHED_PATH_DEPTH, HASHED_PATH_SIZE);
+        File existingFile = new File("target/" + binarySubPath + "/" + uuid);
```

Comment:
please use:
```
@Rule
    public final TemporaryFolder tmpFolder = new TemporaryFolder();
```
to create a temp directory and put the file into there. The target directory isn't too bad, but won't get automatically cleaned up (in case the test fails) without a `mvn clean`